core/state_processor.go

```diff
@@ -141,8 +141,16 @@ func applyTransaction(config *params.ChainConfig, gp *GasPool, statedb *state.In
 // indicating the block was invalid.
 func ApplyTransaction(config *params.ChainConfig, getHeader func(hash common.Hash, number uint64) *types.Header, engine consensus.Engine, author *common.Address, gp *GasPool, ibs *state.IntraBlockState, stateWriter state.StateWriter, header *types.Header, tx types.Transaction, usedGas *uint64, cfg vm.Config, contractHasTEVM func(contractHash common.Hash) (bool, error)) (*types.Receipt, []byte, error) {
 	// Create a new context to be used in the EVM environment
-	blockContext := NewEVMBlockContext(header, getHeader, engine, author, contractHasTEVM)
-	vmenv := vm.NewEVM(blockContext, vm.TxContext{}, ibs, config, cfg)
+
+	var vmenv vm.VMInterface
+
+	if tx.IsCairo() {
```

Comment:
probably `tx.IsStarkNet` is better naming, especially if we have a non-cairo based txs in the future

Reply:
Changed